docs/developers/operations-api/components.md

```diff
@@ -285,3 +285,170 @@ _Operation is restricted to super_user roles only_
+- hostname _(required)_ - the hostname for the ssh config (see below). Used to map `host` to an actual domain (e.g. `github.com`)
+- known_hosts _(optional)_ - the public SSH keys of the host your component will be retrieved from. If `hostname` is `github.com` this will be retrieved automatically. Line breaks must be delimited with `\n`
+
+### Body
+
+```json
+{
+  "operation": "add_ssh_key",
+  "name": "harperdb-private-component",
+  "key": "-----BEGIN OPENSSH PRIVATE KEY-----\nthis\nis\na\nfake\nkey\n-----END OPENSSH PRIVATE KEY-----",
+  "host": "harperdb-private-component.github.com",
+  "hostname": "github.com"
+}
+```
+
+### Response: 200
+```json
+{
+  "message": "Added ssh key: harperdb-private-component"
+}
+```
+
+### Generated Config and Deploy Component "package" string examples
+```
+#harperdb-private-component
+Host harperdb-private-component.github.com
+        HostName github.com
+        User git
+        IdentityFile /hdbroot/ssh/harperdb-private-component.key
+        IdentitiesOnly yes
+```
+
+```
+"package": "git+ssh://git@<host>:<github-repo-path>.git#semver:v1.2.3"
+
+"package": "git+ssh://git@harperdb-private-component.github.com:HarperDB/harperdb-private-component.git#semver:v1.2.3"
+```
+
+Note that `deploy_component` with a package uses `npm install` so the url must be a valid npm format url. The above is an example of a url using a tag in the repo to install.
+
+## Update SSH Key
+Updates the private key contents of an existing SSH key.
+
+_Operation is restricted to super_user roles only_
+
+- operation _(required)_ - must always be `update_ssh_key`
+- name _(required)_ - the name of the key to be updated
+- key _(required)_ - the private key contents. Line breaks must be delimited with `\n`
+
+### Body
+```json
+{
+  "operation": "update_ssh_key",
+  "name": "harperdb-private-component",
+  "key": "-----BEGIN OPENSSH PRIVATE KEY-----\nthis\nis\na\nNEWFAKE\nkey\n-----END OPENSSH PRIVATE KEY-----",
+  "host": "harperdb-private-component.github.com",
+  "hostname": "github.com"
+}
+```
+
+### Response: 200
+```json
+{
+    "message": "Updated ssh key: harperdb-private-component"
+}
+```
+
+## Delete SSH Key
+Deletes a SSH key. This will also remove it from the generated SSH config.
+
+_Operation is restricted to super_user roles only_
+
+- operation _(required)_ - must always be `delete_ssh_key`
+- name _(required)_ - the name of the key to be deleted
+
+### Body
+```json
+{
+  "name": "harperdb-private-component"
+}
+```
+
+### Response: 200
+```json
+{
+    "message": "Deleted ssh key: harperdb-private-component"
+}
+```
+
+## List SSH Keys
+List off the names of added SSH keys
+
+_Operation is restricted to super_user roles only_
+
+- operation _(required)_ - must always be `list_ssh_keys`
+
+### Body
+```json
+{
+  "operation": "list_ssh_keys"
+}
+```
+
+### Response: 200
+```json
+[
+    {
+        "name": "harperdb-private-component.key"
```

Comment:
In this example, the name listed here ("harperdb-private-component.key") doesn't match the name of the SSH key in the add_ssh_key ("harperdb-private-component"). Is that typo? Or are they actually inconsistent with each other?

Reply:
This was because of a typo by me in my core PR. I updated that which makes it output without `.key` so i updated this to reflect. Both should be without the `.key`